src/main/java/org/opensearch/knn/indices/ModelIndex.java

```diff
@@ -0,0 +1,229 @@
+import org.opensearch.action.index.IndexRequestBuilder;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Base64;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.opensearch.knn.index.KNNSettings.MODEL_INDEX_NUMBER_OF_REPLICAS_SETTING;
+import static org.opensearch.knn.index.KNNSettings.MODEL_INDEX_NUMBER_OF_SHARDS_SETTING;
+
+/**
+ * ModelIndex is a singleton class that controls operations on the model system index
+ */
+public class ModelIndex {
+    public static Logger logger = LogManager.getLogger(ModelIndex.class);
+
+    //TODO: Should these be stored in constants or in the class? In PR get feedback
+    public static final String MODEL_INDEX_MAPPING = "mappings/model-index.json";
+    public static final String MODEL_INDEX_NAME = ".knn-model-index";
+
+    private int numberOfShards;
+    private int numberOfReplicas;
+
+    private static ModelIndex INSTANCE;
+    private static Client client;
+    private static ClusterService clusterService;
+    private static Settings settings;
+
+    /**
+     * Make sure we just have one instance of model index
+     *
+     * @return ModelIndex instance
+     */
+    public static synchronized ModelIndex getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ModelIndex();
+        }
+        return INSTANCE;
+    }
+
+    public static void initialize(Client client, ClusterService clusterService, Settings settings) {
+        ModelIndex.client = client;
+        ModelIndex.clusterService = clusterService;
+        ModelIndex.settings = settings;
+    }
+
+    private ModelIndex() {
+        numberOfShards = MODEL_INDEX_NUMBER_OF_SHARDS_SETTING.get(settings);
+        numberOfReplicas = MODEL_INDEX_NUMBER_OF_REPLICAS_SETTING.get(settings);
+
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(MODEL_INDEX_NUMBER_OF_SHARDS_SETTING,
+                it -> numberOfShards = it);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(MODEL_INDEX_NUMBER_OF_REPLICAS_SETTING,
+                it -> numberOfReplicas = it);
+    }
+
+    /**
+     * Creates model index. It is possible that the 2 threads call this function simulateously. In this case, one
+     * thread will throw a ResourceAlreadyExistsException. This should be caught and handled.
+     *
+     * @param actionListener CreateIndexResponse listener
+     * @throws IOException thrown when get mapping fails
+     */
+    public void createModelIndex(ActionListener<CreateIndexResponse> actionListener) throws IOException {
```

Comment:
Any reason we don't have to check verify that model doesn't exist before create?

Reply:
So, the modelIndexExists function can be used before calling this function. There is a chance 2 threads try to create the index. In which case, `ResourceAlreadyExistsException` will be thrown and should be handled. 

I think either approach will work. Do you think this function should include the modelIndexExists check as well? Or should user be responsible for calling modelIndexExists first.

Reply:
nit:Create/Update/Delete instead of CreateModelIndex since the class name is ModelIndex

Reply:
> So, the modelIndexExists function can be used before calling this function. There is a chance 2 threads try to create the index. In which case, `ResourceAlreadyExistsException` will be thrown and should be handled.
> 
> I think either approach will work. Do you think this function should include the modelIndexExists check as well? Or should user be responsible for calling modelIndexExists first.

can this be made singleton to avoid race condition ( check is exists? return false: return true )

Reply:
So it can be combined into one function, however, it still would not be thread safe. However, this will not cause an issue as long as the exception is caught. Refer to [AD](https://github.com/opensearch-project/anomaly-detection/blob/cfebf07f637c05753e32a1648d05db98e1120e0f/src/main/java/org/opensearch/ad/transport/handler/AnomalyResultBulkIndexHandler.java#L99).

Also, will update naming.